app/src/main/java/com/ikarabulut/Game.java

```diff
@@ -0,0 +1,36 @@
+package com.ikarabulut;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Game {
+    private Board currentBoard;
+    private Board nextBoard;
+
+    public Game(Board board) {
+        currentBoard = board;
+        nextBoard = new Board(board.getRows(), board.getColumns());
+
+    }
+
+    public List<ArrayList> getNextBoard() {
+        return this.nextBoard.getBoard();
+    }
+
+    public void generateNextBoard() {
+        for (int row = 0; row < currentBoard.getRows(); row++) {
+            for (int column = 0; column < nextBoard.getColumns(); column++) {
+                int currentCellNumOfAliveNeighbors = currentBoard.getNumberOfAliveNeighbors(row, column);
+                boolean currentCellIsAlive = currentBoard.isCellAliveAt(row, column);
+                boolean nextGenerationStatus = Rules.determineFate(currentCellNumOfAliveNeighbors, currentCellIsAlive);
+                if (nextGenerationStatus) {
+                    nextBoard.setAliveAt(row, column);
```

Comment:
I'm trying to think through if Board really needs to know how to `setAliveAt`. You don't have to change in this PR, but we should explore what would happen if we removed that functionality from Board, and Game just inserted an alive/dead cell here rather than asking the Board class to change a Cell's status.

@jramnani

Reply:
I ended up changing to the other implementation. The `generateNextBoard` initializes a new ArrayList and then adds the proper Cell and then swaps the currentBoard with the nextBoard List